lms/djangoapps/certificates/tests/test_webview_views.py

```diff
@@ -172,20 +175,129 @@ def test_linkedin_share_url(self):
+    @ddt.data(
+        (False, False, False),
+        (False, False, True),
+        (False, True, True),
+        (True, True, True),
+        (True, True, False),
+    )
+    @ddt.unpack
+    def test_social_sharing_availablity_microsites(self, facebook_sharing, twitter_sharing, linkedin_sharing):
+        """
+        Test: Facebook, Twitter and LinkedIn sharing availability for microsites.
+        """
+        self._add_course_certificates(count=1, signatory_count=1, is_active=True)
+        test_url = get_certificate_url(course_id=self.cert.course_id, uuid=self.cert.verify_uuid)
+        social_sharing_settings = dict(
+            CERTIFICATE_FACEBOOK=facebook_sharing,
+            CERTIFICATE_TWITTER=twitter_sharing,
+            CERTIFICATE_LINKEDIN=linkedin_sharing,
+        )
+        with patch("django.conf.settings.MICROSITE_CONFIGURATION", {
+            "test_microsite": dict(
+                settings.MICROSITE_CONFIGURATION['test_microsite'],
+                SOCIAL_SHARING_SETTINGS=social_sharing_settings,
+            )
+        }):
+            response = self.client.get(test_url, HTTP_HOST=settings.MICROSITE_TEST_HOSTNAME)
+            self.assertEqual(response.status_code, 200)
+            self.assertEqual("Post on Facebook" in response.content, facebook_sharing)
+            self.assertEqual("Share on Twitter" in response.content, twitter_sharing)
+            self.assertEqual("Add to LinkedIn Profile" in response.content, linkedin_sharing)
+
+    @override_settings(FEATURES=FEATURES_WITH_CERTS_ENABLED)
+    def test_facebook_default_text_microsites(self):
+        """
+        Test: Facebook sharing default text for microsites.
+        """
+        self._add_course_certificates(count=1, signatory_count=1, is_active=True)
+        test_url = get_certificate_url(course_id=self.cert.course_id, uuid=self.cert.verify_uuid)
+        facebook_text = "Facebook text on Test Microsite"
+        social_sharing_settings = dict(
+            CERTIFICATE_FACEBOOK=True,
+            CERTIFICATE_FACEBOOK_TEXT=facebook_text,
+        )
+        with patch("django.conf.settings.MICROSITE_CONFIGURATION", {
+            "test_microsite": dict(
+                settings.MICROSITE_CONFIGURATION['test_microsite'],
+                SOCIAL_SHARING_SETTINGS=social_sharing_settings,
+            )
+        }):
+            response = self.client.get(test_url, HTTP_HOST=settings.MICROSITE_TEST_HOSTNAME)
+            self.assertContains(response, facebook_text)
+
+    @override_settings(FEATURES=FEATURES_WITH_CERTS_ENABLED)
+    def test_twitter_default_text_microsites(self):
+        """
+        Test: Twitter sharing default text for microsites.
+        """
+        self._add_course_certificates(count=1, signatory_count=1, is_active=True)
+        test_url = get_certificate_url(course_id=self.cert.course_id, uuid=self.cert.verify_uuid)
+        twitter_text = "Twitter text on Test Microsite"
```

Comment:
if `Test Microsite` here refers to platform name than I think we should use `"Twitter text on {platform_name}".format(platform_name=settings.MICROSITE_CONFIGURATION['test_microsite']['platform_name'])`